### PR TITLE
SI-5017 Poor performance of :+ operator on Arrays

### PR DESCRIPTION
**Pull request by jedesah**
Control performance of :+ and +: operator on my machine were 700-800 ms

After adding size hint on the implementation in SeqLike, it went down to 500-600 ms

But with specialixed implementation in ArrayOps, brings it down to 300-400 ms

Unfortunatly, this method will only be called when the Array object is being referenced directly as it's type, but that should be the case enough times to justify the extra method.

I ended up removing the sizeHint in SeqLike because it made the execution of the "benchmark" slower when the Array was being manipulated as a Seq.

Side note: Interestingly enough, the benchmark performed better on my virtualized Fedora 17 with JDK 7 than natively on Mac OS X with JDK 6

Review by @axel22

### CODE REVIEW

File: src/library/scala/collection/mutable/ArrayOps.scala [hunk 1]
```diff
@@ -52,6 +52,20 @@ trait ArrayOps[T] extends Any with ArrayLike[T, Array[T]] with CustomParalleliza
...
+    result(repr.length) = elem
+    result
+  }
+
+  def +:[B >: T: scala.reflect.ClassTag](elem: B): Array[B] = {
+    val result = Array.ofDim[B](repr.length + 1)
+    result(0) = elem
+    Array.copy(repr, 0, result, 1, repr.length)
+    result
+  }
```
Comment (retronym) [inline src/library/scala/collection/mutable/ArrayOps.scala:67]: Is this code called from any existing tests? (Replace it with `???` and run partest to see.)

If so, mention that test in the commit comment; if not, please add a test.

**Comment by axel22** [comment]
After addressing the issue of missing tests, LGTM.

**Comment by adriaanm** [comment]
please use the SI-5017 naming convention

**Comment by jedesah** [comment]
Sorry for the delay. Got caught up finishing work stuff.

**Comment by JamesIry** [comment]
Looks like all the discussion has been addressed and LGTM

**Pull request merged**